src/mongocxx/exception/error_code.hpp

```diff
@@ -31,6 +31,92 @@ enum class error_code : std::int32_t {
     k_unknown_write_concern,
 };
 
+MONGOCXX_API const std::error_code& bson_invalid_error();
```

Comment:
I think I like this better than requiring the user to handle libmongoc error codes and domains.  Also, if as Jesse says they are planning to change the C driver's error primitives, this approach would be a safer bet.